### PR TITLE
Add @grafbase/houdini and example

### PR DESCRIPTION
**Pull request by AlecAivazis**
# Description

This PR adds a new package to the monorepo: `@grafbase/houdini` 🥳 

This package exports a [Codegen Plugin](https://www.houdinigraphql.com/api/codegen-plugins) for [Houdini](https://houdinigraphql.com) that currently gives two things:
- support for `@live` queries 
- scalar definitions (generated types and marshaling functions)

There are still some missing pieces. For example, the project's README is pretty much empty and the sveltekit-houdini example only contains a single query that I was using for testing.

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [x] 📖 Requires documentation update

### CODE REVIEW

**Comment by notrab** [comment]
This is so cool @AlecAivazis! We're almost there.

I've tagged @vitalikda for review also. 

I'll catch up with you on Slack about the README and create a guide that points to this as well as update the docs for Live Query integrations.

🚀 🚀 🚀

File: examples/sveltekit-houdini/.env.example [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Refer to the README on how to fill these variables
+PUBLIC_GRAFBASE_API_URL=http://localhost:5173/graphql
```
Comment (vitalikda) [inline examples/sveltekit-houdini/.env.example:2]: By default, our CLI uses 4000 port: 
```suggestion
PUBLIC_GRAFBASE_API_URL=http://localhost:4000/graphql
```

File: examples/sveltekit-houdini/src/routes/+page.gql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+query ExampleQuery @live {
+	todoList(by: { id: "todolist_01GRHTKYNGGZCHSRTE9CP7T591" }) {
```
Comment (vitalikda) [inline examples/sveltekit-houdini/src/routes/+page.gql:2]: I would fetch collection and you no longer need to hard code ID: 
```
query ExampleQuery @live {
	todoListCollection(first: 100) {
		edges {
			node {
				title
			}
		}
	}
}
```

File: examples/sveltekit-houdini/src/client.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { HoudiniClient } from '$houdini';
+import { PUBLIC_API_URL, PUBLIC_API_KEY } from '$env/static/public'
```
Comment (vitalikda) [inline examples/sveltekit-houdini/src/client.ts:2]: `.env.example` has different keys 
 ```suggestion
import { PUBLIC_GRAFBASE_API_URL, PUBLIC_GRAFBASE_API_KEY } from '$env/static/public'
```

**Review by vitalikda — COMMENTED** [review]
Looks great! Thank you 🙌 

I see this error in logs, any ideas why?
```
Error: Not found: /graphql
    at resolve (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:396:13)
    at resolve (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:237:5)
    at #options.hooks.handle (/node_modules/@sveltejs/kit/src/runtime/server/index.js:41:55)
    at Module.respond (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:234:40)
⚠️  Couldn't pull your latest schema: Unexpected token < in JSON at position 0
```

File: examples/sveltekit-houdini/grafbase/schema.graphql [hunk 4]
```diff
@@ -0,0 +1,11 @@
+type TodoList @model {
+	id: ID!
```
Comment (fbjork) [inline examples/sveltekit-houdini/grafbase/schema.graphql:2]: should tweak indentation here

**Comment by AlecAivazis** [comment]
> Error: Not found: /graphql
>     at resolve (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:396:13)
>     at resolve (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:237:5)
>     at #options.hooks.handle (/node_modules/@sveltejs/kit/src/runtime/server/index.js:41:55)
>     at Module.respond (/node_modules/@sveltejs/kit/src/runtime/server/respond.js:234:40)
> ⚠️  Couldn't pull your latest schema: Unexpected token < in JSON at position 0

I suspect this issue is because of the envar mismatch i just fixed - could you pull it down and try again?

File: examples/sveltekit-houdini/src/routes/+page.gql (hunk 2 above)
Comment (AlecAivazis) [reply examples/sveltekit-houdini/src/routes/+page.gql:2]: Thanks for the input! I wanted to expand the example to look more like the urql and relay ones so this file wasn't really meant to be a good example, just something I could use to confirm the behavior

File: examples/sveltekit-houdini/package.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+		"svelte-check": "^2.9.2",
+		"tslib": "^2.4.1",
+		"typescript": "^4.9.3",
+		"vite": "^4.0.0"
+	},
+	"type": "module",
+	"dependencies": {
+		"@zerodevx/svelte-toast": "^0.8.2",
+		"dotenv": "^10.0.0",
+		"jsonpatch": "^3.1.0",
```
Comment (notrab) [inline examples/sveltekit-houdini/package.json:33]: Is this needed in the example?

**Comment by notrab** [comment]
@AlecAivazis I've gone ahead and updated the `README` as well as the example.

I updated the example to use a Svelte store to track the `authToken`:

```ts
import { HoudiniClient } from '$houdini';
import { PUBLIC_GRAFBASE_API_URL } from '$env/static/public';
import { authToken } from '$lib/shared/stores/token';

export default new HoudiniClient({
	url: PUBLIC_GRAFBASE_API_URL,
	fetchParams() {
		return {
			headers: {
				// Token generated by your auth provider: https://grafbase.com/docs/reference/directives#auth
				Authorization: `Bearer ${authToken}`
			}
		};
	}
});
```

The next thing would be to update the `+page.svelte` to include buttons that sign a new JWT (on the client for demo purposes) and write it to the store:

```
<div>
        <button onClick={() => setRole('')}>Set role to public</button>{' '}
        <button onClick={() => setRole('moderator')}>
          Set role to moderator
        </button>{' '}
        <button onClick={() => setRole('admin')}>Set role to admin</button>
      </div>
```

Please let me know if you'd prefer to do that another way. I'm not too familiar with Svelte.

**Comment by notrab** [comment]
Here's what I have in mind so far:

```svelte
<script lang="ts">
	import { SignJWT } from 'jose'
	import { authToken } from '../stores/authtoken';
	import type { PageData } from './$houdini';

	const issuerUrl = 'https://grafbase.com'

	const secret = new Uint8Array(
  ('abc')
    .split('')
    .map((c) => c.charCodeAt(0))
)

const getToken = (role: string) => {
  const groups = role ? [role] : []
  return new SignJWT({ sub: 'user_1234', groups })
    .setProtectedHeader({ alg: 'HS256', typ: 'JWT' })
    .setIssuer(issuerUrl)
    .setIssuedAt()
    .setExpirationTime('2h')
    .sign(secret)
}

const setRole = async (role :string) => {
	const token = await getToken(role)
	 authToken.set(token)
}
	
	export let data: PageData;

	$: ({ GetAllMessages } = data);
</script>

<div>
	<button on:click={() => setRole('')}>Set role to public</button>{' '}
	<button on:click={() => setRole('moderator')}>
		Set role to moderator
	</button>{' '}
	<button on:click={() => setRole('admin')}>Set role to admin</button>
</div>
```

**Review by notrab — APPROVED** [review]
LGTM

**Pull request merged**